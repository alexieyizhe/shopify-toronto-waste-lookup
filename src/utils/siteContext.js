import React from 'react';

export const wasteItems = [
  {
    title: 'Garbage (wrapping and tying)',
    body: [
      'Place item in the <strong>Garbage Bin.</strong>'
    ]
  },
  {
    title: 'Garbage (tying)',
    body: [
      'Place item in the <strong>Garbage (tying) Bin.</strong>'
    ]
  },
  {
    title: 'Garbage (wrapping) ',
    body: [
      'Place item in the <strong>Garbage (wrapping) Bin.</strong>'
    ]
  }
];

export const ItemsContext = React.createContext({
  items: wasteItems,
  favs: new Set(), // TODO: make this into a set
  updateFavs: () => {}
});
