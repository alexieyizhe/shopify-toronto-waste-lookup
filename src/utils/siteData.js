// should refactor into json and move into data folder later

export const siteTheme = {
  colors: {
    primary: '#1d5a94',
    secondary: '#23975e',
    secondaryBg: '#b5d2c4',

    warning: '#F4A867',
    error: '#F07285',

    offBlack: '#333'
  },
  styling: {
    bodySpacing: '1vw',
  }
};

export const siteTitle = 'Toronto Waste Lookup';

export const searchBarEmptyPlaceholder = 'Enter a search';

export const searchResultsPlaceholders = {
  empty: {text: 'No results found. Try changing your search?', color: siteTheme.colors.warning},
  ready: {text: 'Search for a waste item above!', color: siteTheme.colors.secondary},
  waiting: {text: 'Fetching latest data from the Toronto Waste Wizard...', color: 'grey'},
  error: {text: 'An error occurred while fetching data. Please try refreshing the page.', color: siteTheme.colors.error},
  searching: {text: 'Searching for matching items...', color: siteTheme.colors.secondary},
}

export const favsEmptyPlaceholder = 'You have no favourites. Try adding some with the star icon next to a waste item\'s name!';

export const siteSubtitle = 'Favourites';

export const wasteDataAPIEndPoint = 'https://secure.toronto.ca/cc_sr_v1/data/swm_waste_wizard_APR?limit=1000';

export const FetchStateEnum = { ERROR: 0, WAITING: 1, READY: 2, SEARCHING: 3 };

export const LocalStorageKeysEnum = {
  lastAPIResults: 'wasteItems',
  lastSearchQuery: 'searchQuery',
  lastSearchResults: 'searchResults',
  lastFavourites: 'currentFavs',
  lastStatus: 'appStatus',
};
