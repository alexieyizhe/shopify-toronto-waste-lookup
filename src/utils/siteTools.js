import { css } from 'styled-components';

const displaySizes = {
  large: 1024,
  mid: 832,
  small: 640
};

// Create media template for responsive designs
export const mediaSize = Object.keys(displaySizes).reduce((acc, label) => {
  acc[label] = (...args) => css`
    @media (min-width: ${displaySizes[label] / 16}em) {
      ${css(...args)};
    }
  `;

  return acc;
}, {});
