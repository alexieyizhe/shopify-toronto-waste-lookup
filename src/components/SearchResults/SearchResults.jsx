import React from 'react';
import styled from 'styled-components';

import { resultsEmptyPlaceholder } from '../../utils/siteData';


const ComponentContainer = styled.div`
  height: auto;
  min-height: 40vh;

  margin: ${props => props.theme.styling.bodySpacing}; // margin allows for margin collapsing
`;

const EmptyDisclaimer = styled.div`
  width: 100%;
  padding: ${props => props.theme.styling.bodySpacing};
  color: grey;
  text-align: center;
`;

// Component
export default ({ children }) => (
  <ComponentContainer>
    {children.length > 0 ? children : <EmptyDisclaimer><span>{resultsEmptyPlaceholder}</span></EmptyDisclaimer>}
  </ComponentContainer>
);
