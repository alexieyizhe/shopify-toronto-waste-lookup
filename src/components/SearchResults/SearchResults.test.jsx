import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { ThemeProvider } from 'styled-components';

import { ItemsContext } from '../../utils/siteContext';
import { siteTheme, FetchStateEnum } from '../../utils/siteData';


import SearchResults from './SearchResults';


/* eslint-disable no-undef */
ItemsContext.Consumer = jest.fn((props) => props.children({ appStatus: FetchStateEnum.ERROR }));

it.skip('renders correctly', () => { // skip test becase react-pose needs ref to be forwarded, limitations with jest don't allow for this (https://popmotion.io/blog/20181101-react-pose-4-forwardref/)
  const tree = renderer.create(
    <ThemeProvider theme={siteTheme}>
      <SearchResults>
        <div key="testDiv">test div</div>
        <a  key="testLinkA" href="test thing thing">test link</a>
        <div key="testDiv2">
          <div>
            wogewog
            <span>
              wgwegww????!@$!@
            </span>
          </div>
        </div>
      </SearchResults>
    </ThemeProvider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
