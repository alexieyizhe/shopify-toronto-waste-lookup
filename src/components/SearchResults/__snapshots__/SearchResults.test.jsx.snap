// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  height: auto;
  min-height: 40vh;
  margin: 1vw;
}

<div
  className="c0"
/>
`;
